src/elements/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useRef, useState } from "react"
+import styled from "styled-components"
+import { color } from "../../helpers"
+import { Flex } from "../Flex"
+import { Sans } from "../Typography"
+import { Bar } from "./Bar"
+import { ProvideMousePosition } from "./MousePositionContext"
+
+const ChartContainer = styled(Flex)`
+  border-bottom: 1px solid ${color("black10")};
+  flex: 1;
+`
+
+const useHasEnteredViewport = (ref: React.RefObject<HTMLElement>) => {
+  const [hasEntered, setHasEntered] = useState(false)
+  useEffect(() => {
+    const handleScroll = () => {
+      const rect = ref.current.getBoundingClientRect()
+      if (rect.top <= window.innerHeight - rect.height) {
+        setHasEntered(true)
+        window.removeEventListener("scroll", handleScroll)
+      }
+    }
+    window.addEventListener("scroll", handleScroll)
+    window.dispatchEvent(new Event("scroll"))
+    return () => {
+      window.removeEventListener("scroll", handleScroll)
+    }
+  })
+  return hasEntered
+}
```

Comment:
Not quite ready to start moving these into a more general place (since it's not being used anywhere else) but def something to think about in the future.